### PR TITLE
RDRP-313: TMI Step 5 - Calculate headcount and employment totals

### PR DESCRIPTION
**Pull request by JenCheshire**
- `calculate_totals()` added to `tmi_imputation.py`  to calculate headcount and employment totals
- Function runs at the end of `run_tmi()`
- selectiontype as an argument on `run_short_to_long()` so it can be specified in pipeline

Closes RDRP-313
Closes RDRP-476

#### Code

- [ ] **Code runs** The code runs on my machine and/or CDSW
- [ ] **Conflicts resolved** There are no conflicts (I have performed a rebase if necessary)
- [ ] **Requirements** My/our code functions according to the requirements of the ticket
- [ ] **Dependencies** I have updated the environment yaml so it includes any new libraries I have used
- [ ] **Configuration file updated** any high level parameters that the user may interact with have been put into the config file (and imported to the script)
- [ ] **Clean Code**
    - [ ] Code is as [PEP 8]([url](https://peps.python.org/pep-0008/)) compliant as I can humanly make it
    - [ ] Code passess flake8 linting check
    - [ ] Code adheres to [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)
- [ ] **Type hints** All new functions have [type hints ](https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html)

#### Documentation

Any new code includes all the following forms of documentation:

- [ ] **Function Documentation** Docstrings within the function(s')/methods have been created
    - [ ] Includes `Args` and `returns` for all major functions
    - [ ] The docstring details data types
- [ ] **Updated Documentation**: User and/or developer working doc has been updated

#### Data
- [ ] All data needed to run this script is available in Dev/Test
- [ ] All data is excluded from this pull request
- [ ] Secrets checker pre-commit passes

#### Testing
- [ ] **Unit tests** Unit tests have been created and are passing _or a new ticket to create tests has been created_

---

# Peer Review Section

- [ ] All requirements install from (updated) `environment.yaml`
- [ ] Documentation has been created and is clear - **check the working document**
- [ ] Doctrings ([Google format](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html)) have been created and accurately describe the function's functionality
- [ ] Unit tests pass, or if not present _a new ticket to create tests has been created_
- [ ] **Code runs** The code runs on reviewer's machine and/or CDSW

#### Final approval (post-review)

The author has responded to my review and made changes to my satisfaction.
- [ ] **I recommend merging this request.**

---

### Review comments

*Insert detailed comments here!*

These might include, but not exclusively:

- bugs that need fixing (does it work as expected? and does it work with other code
  that it is likely to interact with?)
- alternative methods (could it be written more efficiently or with more clarity?)
- documentation improvements (does the documentation reflect how the code actually works?)
- additional tests that should be implemented (do the tests effectively assure that it
  works correctly?)
- code style improvements (could the code be written more clearly?)
- Do the changes represent a change in functionality so the version number should increase? Start a discussion if so.
- As a review you can generates the same outputs from running the code

Your suggestions should be tailored to the code that you are reviewing.
Be critical and clear, but not mean. Ask questions and set actions.

### CODE REVIEW

File: src/imputation/tmi_imputation.py [hunk 1]
```diff
@@ -439,9 +459,11 @@ def run_tmi(full_df, target_variables, sic_mapper):
 
     df = pd.concat([final_tmi_df, shortform_df])
 
-    final_df = df.sort_values(
-        ["reference", "instance"], ascending=[True, True]
-    ).reset_index(drop=True)
+    df = df.sort_values(["reference", "instance"],
+                        ascending=[True, True]).reset_index(drop=True)
```
Comment (AnneONS) [inline src/imputation/tmi_imputation.py]: the formatting in the deleted code is correct:
df = df.sort_values(
      etc
).etc

File: src/imputation/tmi_imputation.py [hunk 2]
```diff
@@ -405,6 +405,26 @@ def apply_tmi(df, target_variables, mean_dict):
     return final_df
 
 
+def calculate_totals(df):
```
Comment (AnneONS) [inline src/imputation/tmi_imputation.py:408]: What is needed here is to also keep the original values where imputation does not take place. 
You might like to look at how this is done in the the rest of tmi_imputation.py

**Review by AnneONS — COMMENTED** [review]
need to keep orig values along with imputed updates

File: src/imputation/tmi_imputation.py (hunk 2 above)
Comment (AnneONS) [reply src/imputation/tmi_imputation.py:408]: Ignore this, sorry, it's already done

File: src/imputation/imputation_main.py [hunk 3]
```diff
@@ -35,7 +35,7 @@ def run_imputation(
 
     df = run_apportionment(df)
 
-    df = run_short_to_long(df)
+    df = run_short_to_long(df, selectiontype=["C"])
```
Comment (AnneONS) [inline src/imputation/imputation_main.py]: This actually now needs to be selectiontype=["C", "P"]

File: src/imputation/tmi_imputation.py (hunk 1 above)
Comment (JenCheshire) [reply src/imputation/tmi_imputation.py]: Changed

File: src/imputation/imputation_main.py (hunk 3 above)
Comment (JenCheshire) [reply src/imputation/imputation_main.py]: Changed

**Review by AnneONS — APPROVED** [review]
This is ready to merge :-)

**Pull request merged**